### PR TITLE
Add binary source documentation for starting a build

### PR DESCRIPTION
**Pull request by mfojtik**
This is a documentation part of: https://trello.com/c/oI1ddWEn/273-5-users-can-push-binary-contents-to-trigger-a-build

### CODE REVIEW

**Comment by mfojtik** [comment]
@adellape @smarterclayton PTAL

File: dev_guide/builds.adoc [hunk 1]
```diff
@@ -576,6 +576,28 @@ Specify the `--follow` flag to stream the build's logs in stdout:
...
 ----
 
+Specify the `--env` flag to set environment variable for the build:
+
+----
+$ oc start-build <BuildConfigName> --env=RACK_ENV=development
+----
+
+[[starting-a-build-from-binary-source]]
+Users can push the binary contents into a build rather than relying on the
```
Comment (rhcarvalho) [inline dev_guide/builds.adoc]: Need to tell that the BC must have binary source for that to work?

**Comment by rhcarvalho** [comment]
Let's please queue this after #1289, so that whoever goes to http://docs.openshift.org/latest/welcome/index.html will not have to wonder what is this "binary source" :)

File: dev_guide/builds.adoc (hunk 1 above)
Comment (mfojtik) [inline dev_guide/builds.adoc]: @rhcarvalho it does not need to have binary source for that.

File: dev_guide/builds.adoc [hunk 2]
```diff
@@ -576,6 +576,28 @@ Specify the `--follow` flag to stream the build's logs in stdout:
 $ oc start-build <BuildConfigName> --follow
 ----
 
+Specify the `--env` flag to set environment variable for the build:
+
+----
+$ oc start-build <BuildConfigName> --env=RACK_ENV=development
+----
```
Comment (php-coder) [inline dev_guide/builds.adoc]: Should we update other sections and mention about it? For instance:
- https://docs.openshift.org/latest/dev_guide/builds.html#configuring-the-source-environment
- https://docs.openshift.org/latest/dev_guide/builds.html#docker-strategy-environment
- https://docs.openshift.org/latest/dev_guide/builds.html#custom-strategy-environment

**Comment by adellape** [comment]
@mfojtik Needs a rebase.

**Comment by mfojtik** [comment]
@adellape rebased

**Comment by smarterclayton** [comment]
Saying "binary contents" is probably too strong - generally I would say
"push the source for your build, which could be the contents of a Git or
SVN working directory, a set of prebuilt binary artifacts you wish to
deploy, or a single file".  Binary tends to be too closely associated with
build output, not build input.

File: dev_guide/builds.adoc [hunk 3]
```diff
@@ -975,6 +975,29 @@ Specify the `--follow` flag to stream the build's logs in stdout:
 $ oc start-build <BuildConfigName> --follow
 ----
 
+Specify the `--env` flag to set environment variable for the build:
```
Comment (adellape) [inline dev_guide/builds.adoc]: s/environment variable/any environment variables/

File: dev_guide/builds.adoc [hunk 4]
```diff
@@ -975,6 +975,29 @@ Specify the `--follow` flag to stream the build's logs in stdout:
...
+
+----
+$ oc start-build <BuildConfigName> --env=RACK_ENV=development
```
Comment (adellape) [inline dev_guide/builds.adoc]: s/RACK_ENV=development/<key>=<value>/

File: dev_guide/builds.adoc [hunk 5]
```diff
@@ -975,6 +975,29 @@ Specify the `--follow` flag to stream the build's logs in stdout:
...
+----
+
+[[starting-a-build-from-binary-source]]
+Users can push the binary contents into a build rather than relying on the
+Git source pull or a Dockerfile. This can be done by specifying one of the
+following options for the `start-build` command:
+
+* `--from-dir=<directory>` specifies a directory that will be archived and used as a binary input for the build.
+* `--from-file=<file>` specifies a single file that will be the only file in the build source.
+* `--from-repo=<local source repository>` specifies a path to a local repository to use as the binary input for a build
```
Comment (adellape) [inline dev_guide/builds.adoc]: `<local_source_repository>`

Also missing period.

File: dev_guide/builds.adoc [hunk 6]
```diff
@@ -975,6 +975,29 @@ Specify the `--follow` flag to stream the build's logs in stdout:
...
+
+For example, the following command sends the contents of a local Git repository
+as an archive from the tag 'v2' and starts the build of it:
```
Comment (adellape) [inline dev_guide/builds.adoc]: Use backticks instead of single-quotes.

File: dev_guide/builds.adoc [hunk 7]
```diff
@@ -975,6 +975,29 @@ Specify the `--follow` flag to stream the build's logs in stdout:
...
+
+----
+$ oc start-build hello-world --from-repo=../hello-world --commit=v2
+----
```
Comment (adellape) [inline dev_guide/builds.adoc]: Wrap all of this also in an example block (`====`) since it's showing an example command with real values.

**Comment by adellape** [comment]
@mfojtik Some comments inline. See also above https://github.com/openshift/openshift-docs/pull/1312#issuecomment-164574303 if it was missed previously.

**Comment by adellape** [comment]
@mfojtik bump

**Comment by adellape** [comment]
Closing in favor of https://github.com/openshift/openshift-docs/pull/1583.

**Pull request closed without merging**